src/vm/node_modules/VM.js

```diff
@@ -518,6 +518,40 @@ function getZpools(log, callback)
     });
 }
 
+function getOptimalVolblocksize(log, volPath, callback)
+{
+    assert.object(log, 'no logger passed to getOptimalVolblocksize()');
+    assert.string(volPath, 'no volume path passed to getOptimalVolblocksize()');
+
+    var pool = volPath.split('/')[0];
+    var cmd = '/usr/sbin/zdb';
+    var args = ['-C', pool];
+
+    traceExecFile(cmd, args, log, 'zdb-c',
+        function (error, stdout, stderr) {
+
+        if (error) {
+            log.error('Unable to get zdb pool data');
+            callback(error, {'stdout': stdout, 'stderr': stderr});
+            return;
+        }
+
+        var hasAshift9  = stdout.match('ashift: 9');
+        var hasAshift12 = stdout.match('ashift: 12');
+        var hasRaidZ = stdout.match('type: \'raidz\'');
+
+        // This is an approximation: we only increase the blocksize if we know
+        // that the entire pool is using 4K pages. If even one disk is using
+        // 512b pages (e.g. the data disks are 4K, but the slog is 512b) then
+        // we stick with the default.
```

Comment:
Editoral comment nit: You should mention briefly here about WHY you're increasing the block size to 32k (vs DEFAULT_VOLBLOCKSIZE's 8k).  Suggested additional sentences at the end...

```
        // This is an approximation: we only increase the blocksize if we know
        // that the entire pool is using 4K pages. If even one disk is using
        // 512b pages (e.g. the data disks are 4K, but the slog is 512b) then
        // we stick with the default.  Increasing the blocksize reduces the
        // overhead noticeably on RAIDZ configurations to allow more use
        // of available disk space. Overhead measurements (detailed in
        // OS-8531) suggest 32KiB is a good larger block size.
```

Reply:
Done. Hopefully I didn't expand too much.